src/components/icons/IconsStoriesAll.tsx

```diff
@@ -0,0 +1,209 @@
+		const value = event?.target?.value;
+
+		set_searchInputValue(value ? value : '');
+
+		localStorage.setItem(localStorageSearchText, value);
+		// need to manually dispatch for this picked up by the split view since it's in the same tab
+		window.dispatchEvent(new Event('storage'));
+	}
+
+	function onStorage() {
+		// SELECTED ICON
+
+		const lsIconPropKey = window.localStorage.getItem(localStorageIconPropKey);
+
+		if (typeof lsIconPropKey !== undefined && lsIconPropKey !== null && selectedIconData?.exportIconName !== lsIconPropKey) {
+			const Icon = getIconComponentRef(lsIconPropKey);
+
+			if (!Icon) {
+				return;
+			}
+
+			// clear out all additional prop value changes made
+			set_additionalPropChanges({});
+			// select clicked icon
+			set_selectedIconData({
+				exportIconName: lsIconPropKey,
+				exportNamespaceIconName: getExportNamespaceIconName(lsIconPropKey),
+				Icon,
+				meta: Icon.meta,
+			});
+		}
+
+		// SEARCH TEXT
+
+		const lsSearchText = window.localStorage.getItem(localStorageSearchText);
+
+		if (!!lsSearchText) {
+			set_searchInputValue(lsSearchText);
+		}
+		else {
+			set_searchInputValue('');
+		}
+	}
+
+	Object.entries(additionalPropChanges).map(([propName, propValue]) => {
+		changedAdditionalProps[propName] = propValue;
+	});
+
+	React.useEffect(() => {
+		// check for initial persisted values
+		onStorage();
+
+		window.addEventListener('storage', onStorage);
+
+		return () => {
+			window.removeEventListener('storage', onStorage);
+		}
+	  }, []);
+
+	return (
+		<IconStoriesAllContext.Provider value={{
+			additionalPropChanges,
+			changedAdditionalProps,
+			selectedIconData,
+			set_additionalPropChanges,
+			set_doShowAdditionalProps,
+			doShowAdditionalProps,
+		}}>
+			<div className={styles.IconsStoriesAll}>
+				<div className={styles.IconsStoriesAll_Header}>
+					<InputSearch
+						onChange={onChangeInput}
+						placeholder="Filter by icon names and tags ..."
+						value={searchInputValue}
+					/>
+				</div>
+				<div className={styles.IconsStoriesAll_Content}>
+					{iconsPropKeys.map((iconPropKey) => {
+						// ignore if property key is not a string or is in exclusion list
+						if (typeof iconPropKey !== 'string'
+							|| [
+								'Icons',
+								'__esModule',
+								'Symbol(Symbol.toStringTag)'
+							].includes(iconPropKey as string)
+						) {
+							return;
+						}
+
+						const Icon = getIconComponentRef(iconPropKey);
+						const exportNamespaceIconName = getExportNamespaceIconName(iconPropKey);
+						const meta: IconSvgStoryMeta | undefined = Icon?.meta;
+						const IconRef: React.FC = Icon as React.FC;
+
+						// filter out non-matching icons if there is search criteria
+						if (searchInputValue
+							&& !iconPropKey.toLowerCase().includes(searchInputValue.toLowerCase())
+							&& !exportNamespaceIconName?.toLowerCase().includes(searchInputValue.toLowerCase())
+							&& !meta?.tags?.join('`').toLowerCase().includes(searchInputValue.toLowerCase())
```

Comment:
What do you think about something like?
```suggestion
!meta?.tags?.map(tag => tag.toLowerCase()).includes(searchInputValue.toLowerCase())
```

Reply:
Just tried it and it doesn't catch all search tags like the original does 🤔 I'll have to think through this again to see what it's doing and whether I can write something a little more intuitive as to what's going on and why.

Reply:
Totally! I'm cool with it as is and we can also iterate and improve in the future! :rocket: